node/src/actors/session/handlers.rs

```diff
@@ -636,51 +647,17 @@ fn session_last_beacon_inbound(
 
 fn session_last_beacon_outbound(
     session: &Session,
-    ctx: &mut Context<Session>,
-    CheckpointBeacon {
-        checkpoint: received_checkpoint,
-        ..
-    }: CheckpointBeacon,
+    _ctx: &mut Context<Session>,
+    beacon: CheckpointBeacon,
 ) {
-    // Get ChainManager address from registry
-    let chain_manager_addr = System::current().registry().get::<ChainManager>();
-    // Send GetHighestCheckpointBeacon message to ChainManager
-    chain_manager_addr
-        .send(GetHighestCheckpointBeacon)
-        .into_actor(session)
-        .then(move |res, _act, ctx| {
-            match res {
-                Ok(Ok(chain_beacon)) => {
-                    if chain_beacon.checkpoint > received_checkpoint {
-                        warn!(
-                            "My outbound peer is behind me (Outbound checkpoint:{} < Chain checkpoint:{})",
-                            received_checkpoint, chain_beacon.checkpoint
-                        );
-                    } else if chain_beacon.checkpoint == received_checkpoint {
-                        info!("Our chain is on par with our peer's",);
-                    } else {
-                        debug!(
-                            "Received a checkpoint beacon that is ahead of ours ({} > {})",
-                            received_checkpoint, chain_beacon.checkpoint
-                        );
-                    }
-
-                    // Handle mine boolean
-                    ChainManager::from_registry().do_send(PeerLastEpoch {
-                        epoch: received_checkpoint,
-                    });
-
-                    actix::fut::ok(())
-                }
-                _ => {
-                    warn!("Failed to get highest checkpoint beacon from ChainManager");
-                    // FIXME(#72): a full stop of the session is not correct (unregister should
-                    // be skipped)
-                    ctx.stop();
+    SessionsManager::from_registry().do_send(PeerBeacon {
+        address: session.remote_addr,
+        beacon,
+    })
+}
 
-                    actix::fut::err(())
-                }
-            }
-        })
-        .wait(ctx);
+fn send_last_beacon(session: &mut Session, beacon: CheckpointBeacon) {
```

Comment:
I will add a comment to indicate that it would be cool to choose send to inbounds or outbounds or both